airflow/models.py

```diff
@@ -3671,12 +3671,21 @@ def topological_sort(self):
 
     @provide_session
     def set_dag_runs_state(
-            self, state=State.RUNNING, session=None):
+            self,
+            state=State.RUNNING,
+            session=None,
+            only_include_backfill_dagruns=False):
         drs = session.query(DagModel).filter_by(dag_id=self.dag_id).all()
         dirty_ids = []
         for dr in drs:
-            dr.state = state
-            dirty_ids.append(dr.dag_id)
+            if only_include_backfill_dagruns:
```

Comment:
```python
if not only_backfill_dagruns or dr.is_backfill:
    dr.state = state
    dirty_ids.append(dr.dag_id)
```

Reply:
not sure if this is the same:
for backfill: we only want to set the backfill dag_run back to running and not touch daily run(only_backfill_dagruns && dr.is_backfill)
for clear: we only want to touch and reset daily scheduled dag_run (not only_backfill_dagruns && not dr.is_backfill)